spec/std/isa/csr/Smaia/mvip.yaml

```
@@ -0,0 +1,108 @@
+# Copyright (c) 2025 Kallal Mukherjee
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+
+# yaml-language-server: $schema=../../../schemas/csr_schema.json
+
+$schema: "csr_schema.json#"
+kind: csr
+name: mvip
+long_name: Machine Virtual Interrupt Pending
+address: 0x309
+priv_mode: M
+length: MXLEN
+writable: true
+description: |
+  If a bit in mvien is read-only zero, the corresponding bit in mvip should also be read-only zero.
+  Bits corresponding to unimplemented interrupt sources should be read-only zero.
+definedBy: Smaia
+fields:
+  # Fields VSSIP, VSTIP, VSEIP are defined in the RISC-V AIA specification
+  # Section 3.2.2 "Machine Virtual Interrupt Pending Register (mvip)"
+  # These correspond to virtual supervisor software, timer, and external interrupts
+  VSSIP:
+    location: 1
+    long_name: Virtual Supervisor Software Interrupt Pending
+    description: |
+      Virtual supervisor software pending.
+
+      This bit can only be written if the corresponding VSSIE bit in mvien is writable and enabled.
+    type: RW-R
+    reset_value: UNDEFINED_LEGAL
+    definedBy: Smaia
+    sw_write(csr_value): |
+      # Check if corresponding mvien bit is enabled
+      Bits<MXLEN> current_mvien = CSR[mvien].sw_read();
+      if (current_mvien[1] == 1) {
```

```suggestion
      if (CSR[mvien].VSSIE == 1'b1) {
```

...and similar changes in the similar cases below.